### ISSUE TITLE
Throttle ourselves to spare our upstream dependencies

### ISSUE THREAD

**Issue by wheresrhys** [Financial-Times/spoor-enrichment#286]
CAPI1 throttles us at 150 qpm, which is fine for normal operation, though at particularly busy times, or when clearing a severely backed up queue, we will be throttled by mashery. 

Should we self-throttle in some way to prevent data being unenriched by dependencies which struggle to keep up

**Comment by markstephens** [comment]
Or a local cache..

**Comment by wheresrhys** [comment]
Yep good idea. I thought there was already an issue for that. Will alter the title

**Comment by wheresrhys** [comment]
... although there are likely medium - long-term plans to move spoor-enrichment to lambda, in which case  an in memory cache is not an option.

**Comment by markstephens** [comment]
Lambda isn't likely to be cost effective in this situation, worth investigating of course, but might not be worth it, if cost is the main driver to move.

Lambda does have a cache you can use though.. I did some experiments with the tmp filesystem in lambda and it sticks around for a couple of hours.

**Comment by commuterjoy** [comment]
I can see the point of a simple least recently used cache for each HTTP transform we do - perhaps just ~50MB so we avoid repeated hits on popular items.
